### PR TITLE
Add trait for coordinate of mouse

### LINKED ISSUE #8: Add x/y traits to mouse listener
Based on comments in https://github.com/jupyter-widgets/ipywidgets/pull/1727 this should be opt-in (or perhaps it should be an option to turn off the full event information and just set the x/y trait).

### PR DESCRIPTION
**Pull request by mwcraig**
This PR adds the ability to pass the mouse position vi a trait instead a custom event message.

Assuming `e` is an `Event`, setting `e.xy_trait_coords = 'page'` (or any of the other coordinates returned in a mouse event), means that `e.xy` will be updated with the position of the mouse. Any other event can also be attached and generate event messages like usual.

Setting `e.xy_trait_coords = None` turns off the `xy` trait.

A question that should be answered before merging:

- [ ] What should happen if the user requests the `array` coordinates but that is not defined for the view to which the `Event` widget is attached? This can't be checked when `e.xy_trait_coords` is set, because `e.source` may not have been set yet, and even if the source has been set, there may not be a view, which is what generates the array coordinates. Right now there is an error displayed in the javascript console, but most users won't ever see it. 

Minimum working example:

```python
from ipyevents import Event
from ipywidgets import HTML
h = HTML(value = '<h1>Hello world!</h1>')
e = Event()
e.xy_trait_coords = 'client'
e.source = h
h
```

In a subsequent cell `e.xy` will print the mouse position.

@pbugnion -- if you can review this one it would be helpful. I'll add to the documentation before merging, just holding off on that in case you suggest any changes to the interface.

Closes #8

### CODE REVIEW

**Comment by mwcraig** [comment]
Ping @maartenbreddels  and @ejeschke in case you are interested in this feature...

File: ipyevents/events.py [hunk 1]
```diff
@@ -13,6 +13,17 @@ class Event(CoreWidget):
...
+    xy_trait_coords = Unicode(allow_none=True, default=None).tag(sync=True)
+    _xy_trait_coords_allowed = List([
+        None,       # Not tracking mouse x/y
+        'array',    # "natural" coordinates for the widget (e.g. image)
+        'client',   # Relative to the visible part of the web page
+        'offset',   # Relative to the padding edge of widget
+        'page',     # Relative to the whole document
+        'relative', # Relative to the widget
+        'screen'    # Relative to the screen
+    ]).tag(sync=True)
```
Comment (pbugnion) [inline ipyevents/events.py]: Does `_xy_trait_coords_allowed` need to be serialized? Unless I'm missing something, I'm not sure you're using it in the client.

**Comment by pbugnion** [comment]
This looks great. What if I want the mouse position in multiple coordinate systems?

File: js/src/events.ts [hunk 2]
```diff
@@ -93,6 +93,9 @@ class EventModel extends WidgetModel {
             watched_events: [],
             ignore_modifier_key_events: false,
             prevent_default_action: false,
+            xy_trait_coords: null,
```
Comment (pbugnion) [inline js/src/events.ts]: I'm not sure `xy_trait_coords` is the most descriptive name for this. Maybe `xy_coordinate_system`?

File: js/src/events.ts [hunk 3]
```diff
@@ -187,6 +197,14 @@ class EventModel extends WidgetModel {
                     break
             }
         }
+        // Also add listeners to support populating the x/y traits
+        if (this.get('xy_trait_coords')) {
+            let prevent_default = this.get('prevent_default_action')
+            let handler = this._set_xy.bind(this, view)
+            let event = 'mousemove'
```
Comment (pbugnion) [inline js/src/events.ts]: Unless I'm missing something, these should be `const`, not `let`.

Comment (mwcraig) [reply js/src/events.ts]: Is it just that the last one (`event`) that should be `const`, or all three of them?

File: ipyevents/events.py [hunk 4]
```diff
@@ -13,6 +13,17 @@ class Event(CoreWidget):
...
+    xy_coordinate_system = Unicode(allow_none=True, default=None).tag(sync=True)
+    _xy_coordinate_system_allowed = List([
+        None,       # Not tracking mouse x/y
+        'array',    # "natural" coordinates for the widget (e.g. image)
+        'client',   # Relative to the visible part of the web page
+        'offset',   # Relative to the padding edge of widget
+        'page',     # Relative to the whole document
+        'relative', # Relative to the widget
+        'screen'    # Relative to the screen
+    ]).tag(sync=False)
```
Comment (pbugnion) [inline ipyevents/events.py]: This doesn't even need to be a traitlet -- you could just make it a set: 
```
_xy_coordinate_system_allowed = {
    None, 'array', ...
}
```
It doesn't matter that much, but the fact that it's a traitlet implies that you want to somehow react to user changes on it, so it's a bit confusing for the reader.

Comment (mwcraig) [reply ipyevents/events.py]: good point!

**Comment by mwcraig** [comment]
> This looks great. What if I want the mouse position in multiple coordinate systems?

I'm inclined to say multiple listeners...but am open to suggestions for how to handle that case.

**Comment by pbugnion** [comment]
> I'm inclined to say multiple listeners

That makes sense.

**Comment by mwcraig** [comment]
I think I've addressed all of the comments. I'll add some documentation a little later, once #20 is merged (there are substantial changes to the doc notebook in there).

**Comment by mwcraig** [comment]
Thanks for the feedback, merging.

**Pull request merged**